SavingsManagementSystem.Service/Authentication/Implementations/VerificationTokenService.cs

```diff
@@ -14,14 +14,16 @@ public VerificationTokenService(IUnitOfWork unit)
 			_unit = unit;
 		}
 
-		public async Task<VerificationToken> CreateVerificationTokenAsync(string userId, int expiryMinutes)
+		public async Task<VerificationToken> CreateVerificationTokenAsync(int expiryMinutes, string status, string email, string? userId = null)
 		{
 			var createdOn = DateTime.UtcNow;
 			var vToken = new VerificationToken
 			{
 				Id = Guid.NewGuid().ToString(),
 				UserId = userId,
 				Token = Guid.NewGuid().ToString(),
+				Status = status,
+				Email = email,
 				IsUsed = false,
 				CreatedOn = createdOn,
```

Comment:
is VerificationToken inheriting from base entity. if yes then you dont need to set the createdOn. i can agree with setting the Id because you might want to use it when sending mail. but if you aren't going to use it you can also remove it.

Reply:
it doesn't inherit from base entity

Reply:
why not inheritinh from base entity